app/src/components/queue.js

```diff
@@ -0,0 +1,91 @@
+const Bull = require('bull');
+const config = require('config');
+const log = require('npmlog');
+const uuidv4 = require('uuid/v4');
+
+const email = require('./email');
+
+const chesQueue = new Bull('ches', {
+  redis: {
+    host: config.get('redis.host'),
+    password: config.get('redis.password')
+  }
+});
+
+const queue = {
+  /** Adds an email message to the queue
+   *  @param {object} message An email message object
+   *  @param {object} opts Bull job queue options to override default behavior. For more details on this object structure, refer to the JobOpts interface at https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#queueadd.
+   *  @returns {string} A uuid corresponding to the queued message
+   */
+  enqueue: (message, opts = {}) => {
+    const id = uuidv4();
+    chesQueue.add({
+      message: message
+    }, Object.assign(opts, {
+      jobId: id
+    }));
+
+    log.info('enqueue', `Job ${id} enqueued`);
+    return id;
+  },
+
+  /** Cleanup message data upon job completion
+   *  @param {object} job A Bull Queue Job object
+   */
+  onCompleted: async job => {
+    log.info('queue', `Job ${job.id} completed`);
+    await job.update(null); // Scrub out message information on finish
+  },
+
+  /** Log the job error upon encountering an error
+   *  @param {object} job A Bull Queue Job object
+   */
+  onError: async job => {
+    if (typeof job.id !== 'undefined') {
```

Comment:
Should log some error if the job.id IS undefined? (the else case) or is that handled elsewhere

Reply:
I don't think it's handled elsewhere, but on the same token it's not very useful to know that something errored and not have something to correlate back to for reference.

Reply:
True, but (unless this would be called frequently and thus polute the logs) it is pretty helpful operationally to have something logged in generic error cases where something has crashed and the ops person doesn't know why.

Reply:
I'll add in an else logger for now so that at least everything is being logged. Can look into the deeper edge cases as we encounter them.